### PR TITLE
Macro for displaying changed settings

### PR DESCRIPTION
**Pull request by larsjohnsen**
Only shows modules that have been modified in some way, but ignores certain options like arrays, builders, and lists in order to avoid sensitive and mostly unimportant information.

Macros are available  only for /r/RESissues` and /r/Enhancement, and inserts it in a format like this:
```
    RESTips              
       dailyTip                    	true → false
    accountSwitcher      
    betteReddit          
       commentsLinksNewTabs        	false → true
       hideLinkInstant             	false → true
       scoreHiddenTimeLeft         	true → false
       commentCollapseInInbox      	false → true
    commentQuickCollapse 
       scrollOnCollapse            	false → true
    commentStyle         
       continuity                  	false → true
    commentTools         
    filteReddit          
       filterSubredditsFrom        	"everywhere-except-subreddit" → "legacy"
    hover                
    keyboardNav    
       linearScrollStyle               	"legacy" → "adopt"      
       nonLinearScrollStyle        	"legacy" → "adopt"
       useGoMode                   	true → false
       followLinkNewTabFocus       	true → false
    multiredditNavbar    on → off
    newCommentCount      
       cleanComments               	"7" → 14
       subscriptionLength          	"2" → 7
    nightMode            
    noParticipation      
       disableCommentTextarea      	false → true
    notifications        
    orangered            
       showFloatingEnvelope        	true → false
    penaltyBox           
    quickMessage         on → off
    saveComments         on → off
    selectedEntry        
       addFocusBGColor             	true → false
       addFocusBorder              	true → false
    showImages           
       bufferScreens               	"2" → "1"
       maxHeight                   	"80%" → "90%"
       clippy                      	true → false
       captionsPosition            	"titleAbove" → "creditsBelow"
       markSelftextVisited         	false → true
       galleryRememberWidth        	true → false
       convertGifstoGfycat         	false → true
    showKarma            on → off
    singleClick          on → off
    stylesheet           
    subredditManager     on → off
    userHighlight        
       highlightFirstCommenter     	false → true
    userTagger           
       showTaggingIcon             	true → false
    voteEnhancements     
```

Relevant issue: #4245
Tested in browser: Chrome 59

### CODE REVIEW

**Comment by andytuba** [comment]
This is even better than I envisioned!  Hooray for delivering on old requests.

Could this also include the standard r/resissues diagnostics info: res/browser version, nightmode beta etc?

File: lib/core/options/options.js [hunk 1]
```diff
@@ -80,3 +81,47 @@ export function set(opaqueId: OpaqueModuleId, optionKey: string, value: mixed) {
...
+
+	let optionMaxLength = 0;
+	let moduleMaxLength = 0;
+	for (const { moduleID, modifiedOptions, moduleStatus } of getModified()) {
+		moduleMaxLength = Math.max(moduleMaxLength, moduleID.length);
+		optionMaxLength = Math.max(optionMaxLength, ...Object.keys(modifiedOptions).map(v => v.length));
+
+		lines.push(() =>
+			[
+				`    ${_.padEnd(moduleID, moduleMaxLength)} ${moduleStatus}`,
```
Comment (erikdesjardins) [inline lib/core/options/options.js]: Should be able to use native `.padEnd` http://kangax.github.io/compat-table/es2016plus/#test-String_padding

File: lib/core/options/options.js [hunk 2]
```diff
@@ -80,3 +81,47 @@ export function set(opaqueId: OpaqueModuleId, optionKey: string, value: mixed) {
...
+				...Object.entries(modifiedOptions)
+					.filter(([, { type }]) => types.includes(type))
+					.map(([key, value]) => (
+						`      ${_.padEnd(key, optionMaxLength)}\t${JSON.stringify(value.default)} → ${JSON.stringify(value.value)}`
```
Comment (erikdesjardins) [inline lib/core/options/options.js]: Ditto

File: lib/modules/commentTools.js [hunk 3]
```diff
@@ -672,8 +674,16 @@ function addButtonToMacroGroup(groupName, button) {
 	group.dropdown.append($('<li>').append(button));
 }
 
+function getDebugMacros() {
+	if (!['Enhancement', 'RESissues'].includes(currentSubreddit())) return [];
```
Comment (erikdesjardins) [inline lib/modules/commentTools.js]: I could imagine someone changing `currentSubreddit()` to return lowercase (or _maybe_ there are situations where the URL isn't canonicalized); might as well make this `isCurrentSubreddit('Enhancement') || isCurrentSubreddit('RESissues')`, or make `isCurrentSubreddit` variadic like `isPageType`

File: lib/core/options/options.js (hunk 1 above)
Comment (erikdesjardins) [inline lib/core/options/options.js]: Ah, e3c946b6c8441ad56d6e553fd720999ea7ef6cbe

**Pull request merged**